### DISCUSSION TITLE
docker-mailserver setup on kubernetes

### DISCUSSION THREAD

**Discussion by ErikEngerd** [docker-mailserver/docker-mailserver#2750] [General]
Hi all,

I did a write-up of my setup of docker mailserver on my [website](https://brakkee.org/site/index.php/2022/08/26/migrating-a-mailserver-over-to-k8s/). Also wrote about migrating the e-mails themselves. 

If you have any comments, please let me know. It is working fine AFAICT. I also have roundcube webmail working, deployed in a separate pod, about which I will do a write-up within a couple of days. After that, I will migrate mailman. 

cheers
  Erik

**Comment by polarathene** [comment]
If you like, you can [contribute your write-up to our docs](https://github.com/docker-mailserver/docker-mailserver/blob/master/docs/content/examples/tutorials/blog-posts.md?plain=1), it will appear [on this page](https://docker-mailserver.github.io/docker-mailserver/edge/examples/tutorials/blog-posts/).

**Reply by ErikEngerd** [reply]
I have now completed the write-up for roundcube integration on k8s.

**Comment by ErikEngerd** [comment]
I have created a [pull request](https://github.com/docker-mailserver/docker-mailserver/pull/2760) for this.

**Comment by ErikEngerd** [comment]
I now have mailman 3 fully working on Kubernetes in combination with docker mailserver. Also migrate my existing mailing lists to the new setup. This was certainly the most time consuming part of the migration because of the testing and mainly because of a big fight to get DKIM working. Also, I found several inconsistencies in the documentation that I will report later to the mailman docker project. I expect to do a write up of before next week.

**Reply by ErikEngerd** [reply]
The write-up for mailman is online now.

**Reply by winston0410** [reply]
@ErikEngerd I am trying to setup docker mailserver in k8s as well, I did and it running, but I don't understand where should I run the `./setup.sh`. Have you used that script? Do I need to `kubectl exec -n mailserver --stdin --tty  mailserver-- /bin/sh` go into the container shell to run it and get value (e.g. create user, set dkim)?

**Reply by winston0410** [reply]
I think I have found the answer at the bottom of your article! Thank you so much!!

**Comment by winston0410** [comment]
One more question, have you set up an `initContainers`? When I run `kubectl -n mailserver exec -it deploy/mailserver -- setup email add <email>`, it will returns the followings:

```sh
touch: cannot touch '/tmp/docker-mailserver/postfix-accounts.cf': No such file or directory
/usr/local/bin/helpers/lock.sh: line 28: /tmp/docker-mailserver/addmailuser.lock: No such file or directory
grep: /tmp/docker-mailserver/postfix-accounts.cf: No such file or directory
```

I guess I have to use an initContainers to touch them for me?

**Reply by ErikEngerd** [reply]
I haven't seen this issue. Also not using init containers. Are your volumes setup correctly? The directory /tmp/docker-mailserver is where the config volume is mounted. Looks like it is not mounted.
